src/Microsoft.ComponentDetection.Contracts/FileComponentDetectorWithCleanup.cs

```diff
@@ -56,80 +56,78 @@ protected async Task WithCleanupAsync(
         // determining the files that exist as there will be no subsequent cleanup process.
         if (this.FileUtilityService == null
             || this.DirectoryUtilityService == null
-            || !this.TryGetCleanupFileDirectory(processRequest, out var fileParentDirectory))
+            || !this.TryGetCleanupFileDirectory(processRequest, out var fileParentDirectory)
+            || !cleanupCreatedFiles)
```

Comment:
can we add a unit test : return early if the `cleanupCreatedFiles` is false.

Reply:
That flag was covered to ensure that its not cleaning up if that flag is set to false, added an extra check to ensure that we don't even try to get files/directories though